lib/install.js

```diff
@@ -536,8 +545,21 @@ Installer.prototype.runPreinstallTopLevelLifecycles = function (cb) {
   var steps = []
   var trackLifecycle = this.progress.runTopLevelLifecycles
 
+  var p = {}
+
+  try {
+    p = require(path.join(this.where, 'package.json'))
+  } catch (err) {
+    p = require(process.cwd(), 'package.json')
+  }
+
+  // Construct initial node for top of the tree
+  var rootNode = {
+    package: p
+  }
+
   steps.push(
-    [doOneAction, 'preinstall', this.idealTree.path, this.idealTree, trackLifecycle.newGroup('preinstall:.')]
+    [doOneAction, 'preinstall', process.cwd(), rootNode, trackLifecycle.newGroup('preinstall:.')]
```

Comment:
`process.cwd()` isn't safe for this, as you can run `npm install` in a subdir of a project and it'll install at the project root.